pkgs/by-name/db/dbvisualizer/package.nix

```diff
@@ -0,0 +1,73 @@
+{
+  lib,
+  copyDesktopItems,
+  fetchurl,
+  makeDesktopItem,
+  makeWrapper,
+  stdenv,
+  temurin-jre-bin-17,
+}:
+
+let
+  pname = "dbvisualizer";
+in
+stdenv.mkDerivation (finalAttrs: {
+  inherit pname;
+  version = "24.2.2";
+
+  src =
+    let
+      underscoreVersion = lib.concatStringsSep "_" (lib.splitString "." finalAttrs.version);
+    in
+    fetchurl {
+      url = "https://www.dbvis.com/product_download/dbvis-${finalAttrs.version}/media/dbvis_linux_${underscoreVersion}.tar.gz";
+      hash = "sha256-/K50c7pqpKQ2/EEk2iPSTT+3IbLRF1cdo8GHzkIXiHQ=";
+    };
+
+  strictDeps = true;
+
+  nativeBuildInputs = [
+    copyDesktopItems
+    makeWrapper
+  ];
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = pname;
+      desktopName = "DB Visualizer";
+      tryExec = "dbvis";
+      exec = "dbvis";
+      icon = "dbvisualizer";
+      categories = [ "Development" ];
+      terminal = false;
+      keywords = [
+        "Database"
+        "SQL"
+      ];
+    })
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    cp -a . $out
+
+    mkdir -p $out/share/icons/hicolor/128x128/apps
+    ln -s $out/resources/images/tool/dbvis-icon128x128.png \
+      $out/share/icons/hicolor/128x128/apps/dbvisualizer.png
+
+    ln -s $out/dbvis $out/bin/
+    wrapProgram $out/bin/dbvis --set INSTALL4J_JAVA_HOME ${temurin-jre-bin-17}
```

Comment:
Should it be symlinked, or simply renamed?

Reply:
That's a good question. In my opinion either can make sense. My reasoning behind symlinking it is that I did not want to modify the layout of the tarball, only add onto it. Maybe that is just silly and it should be `mv`d instead.

Reply:
Is this something I should change or was that more so just asking in general?

Reply:
It's not blocking, just a question — though thinking more about it I think moving makes more sense

Reply:
I should have checked sooner... If I replace `ln -s` with `mv` then the application does not run. Instead it tells me ``Error: could not open `/nix/store/cc5g4sinkrghiacmwsjvgkafwzyq5mv2-dbvisualizer-24.2.4/bin/java9-args'``.

So it is what I initially feared: it uses relative paths from the executable to refer to some files. While I could also move `java9-args` to be under `bin/`, and that does solve the problem.

However, then I start getting `java.lang.ClassNotFoundException`s, which means it definitely uses relative paths for more things.